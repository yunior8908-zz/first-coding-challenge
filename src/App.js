import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import HeaderComponent from './components/headerComponent';
import MainComponent from './components/mainComponent';

function App() {
  return (
    <BrowserRouter>
      <div className="App">
        <HeaderComponent />
        <div className="container">
          <React.Suspense fallback={(
            <div className="spinner-border" role="status">
              <span className="sr-only">Loading...</span>
            </div>
)}
          >
            <MainComponent />
          </React.Suspense>
        </div>
      </div>
    </BrowserRouter>
  );
}

export default App;
