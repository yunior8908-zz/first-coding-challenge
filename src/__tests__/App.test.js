/* eslint-disable no-undef */
import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import App from '../App';
import HeaderComponent from '../components/headerComponent';

describe('<App />', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<App />);
  });

  it('should be render without problem', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should have headerComponent of child', () => {
    expect(wrapper.find(HeaderComponent).exists()).toBeTruthy();
  });
});
