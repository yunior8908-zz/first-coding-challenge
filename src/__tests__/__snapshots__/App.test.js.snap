// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should be render without problem 1`] = `
<BrowserRouter>
  <div
    className="App"
  >
    <HeaderComponent />
    <div
      className="container"
    >
      <Suspense
        fallback={
          <div
            className="spinner-border"
            role="status"
          >
            <span
              className="sr-only"
            >
              Loading...
            </span>
          </div>
        }
      >
        <MainComponent />
      </Suspense>
    </div>
  </div>
</BrowserRouter>
`;
